exporter/exporter_test.go

```diff
@@ -2244,3 +2244,94 @@ func TestProcessSentinelSlaves(t *testing.T) {
 		})
 	}
 }
+
+func httpScrapeWithPasswordRun(t *testing.T, tstName, target string, options Options, wants []string, tstWantStatusCode int) {
+	t.Run(tstName, func(t *testing.T) {
+		e, _ := NewRedisExporter(target, options)
+		ts := httptest.NewServer(e)
+
+		u := ts.URL
+		u += "/scrape"
+		v := url.Values{}
+		v.Add("target", target)
+
+		up, _ := url.Parse(u)
+		up.RawQuery = v.Encode()
+		u = up.String()
+
+		wantStatusCode := http.StatusOK
+		if tstWantStatusCode != 0 {
+			wantStatusCode = tstWantStatusCode
+		}
+
+		gotStatusCode, body := downloadURLWithStatusCode(t, u)
+
+		if gotStatusCode != wantStatusCode {
+			t.Fatalf("got status code: %d   wanted: %d", gotStatusCode, wantStatusCode)
+			return
+		}
+
+		// we can stop here if we expected a non-200 response
+		if wantStatusCode != http.StatusOK {
+			return
+		}
+
+		for _, want := range wants {
+			if !strings.Contains(body, want) {
+				t.Errorf("url: %s    want metrics to include %q, have:\n%s", u, want, body)
+				break
+			}
+		}
+		ts.Close()
+	})
+}
+
+func TestHTTPScrapeWithPasswordFile(t *testing.T) {
+	if os.Getenv("TEST_REDIS_PWD_FILE") == "" {
+		t.Skipf("TEST_REDIS_PWD_FILE not set - skipping")
+	}
+
+	passwordFile := os.Getenv("TEST_REDIS_PWD_FILE")
+	passwordMap, err := LoadPwdFile(passwordFile)
+	if err != nil {
+		t.Fatalf("Test Failed, error: %v", err)
+	}
+
+	if len(passwordMap) == 0 {
+		t.Skipf("Password map is empty -skipping")
+	}
+
+	tstName := "scrape-pwd-file"
+	tstWantStatusCode := 0
+
+	wants := []string{
+		`uptime_in_seconds`,
+		`test_up 1`,
+	}
+
+	for target, pwd := range passwordMap {
+		options := Options{
+			Namespace: "test",
+			Password:  pwd,
+			LuaScript: []byte(`return {"a", "11", "b", "12", "c", "13"}`),
+			Registry:  prometheus.NewRegistry(),
+		}
+		httpScrapeWithPasswordRun(t, tstName, target, options, wants, tstWantStatusCode)
+	}
+
+	tstName = "scrape-pwd-file-error-password"
+	wants = []string{
```

Comment:
Instead of redeclaring `wants` here, just make it part of a tests array and then iterate over it.
See [here](https://github.com/oliver006/redis_exporter/blob/master/exporter/exporter_test.go#L1544-L1563) for an example.
This also means you won't need `httpScrapeWithPasswordRun`.